aten/src/ATen/native/cuda/Sorting.cu

```diff
@@ -65,25 +65,34 @@ __global__ void gatherKthValue(
       &kValue);
 
   // Find the index of the k-th highest element
-  index_t kValueIndex = 0;
-  bool foundKValue = false;
+  __shared__ int32_t minIndexFound;
```

Comment:
this (and atomic below) won't correctly work when the kernel is launched with `index_t = int64_t`, for large slices that require 64-bit indexing

Reply:
Thanks for checking! Can you please error out in this case then? Does instantiating with index_t = int64_t ever make sense, or will it only lead to silent wrong results?